src/Features/Core/Portable/QuickInfo/QuickInfoUtilities.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.DocumentationComments;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Tags;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.QuickInfo
+{
+    internal static class QuickInfoUtilities
+    {
+        public static async Task<QuickInfoItem> CreateQuickInfoItemAsync(
+            Workspace workspace,
+            ISymbolDisplayService descriptionService,
+            SemanticModel semanticModel,
+            TextSpan span,
+            ImmutableArray<ISymbol> symbols,
+            SupportedPlatformData? supportedPlatforms = null,
+            bool showAwaitReturn = false,
+            NullableFlowState flowState = NullableFlowState.None,
+            CancellationToken cancellationToken = default)
```

Comment:
@mgoertz-msft you can ignore the supportedPlatform/showAwaitReturn/flowState as i imagine you don't care about them.